### ISSUE TITLE
Aurelia

### ISSUE THREAD

**Issue by beercsaba** [Retyped/Retyped#3]
After reviewing some github issues and conversations it seems, that typescript definition generated by Aurelia itself does not conforms the requirements defined by DefinitelyTyped, so it can not be included into it. I did not dig into the details, it's beyond our abilities. But i've found, that there are official TS definitions for DefinitelyTyped [here](https://www.nuget.org/packages/aurelia-framework.TypeScript.DefinitelyTyped) . 
And there are also nuget packages for the whole Aurelia ecosystem named like aurelia*TypeScript.DefinitelyTyped.

Could you please include these packages into the following release of Retyped? Or else could you generate a custom Retyped package?

**Comment by AndreyChechel** [comment]
Hi @beercsaba, thanks for your interest in Bridge.NET and Retyped!

Below is the list of libraries that are being installed for a project created through **aurelia-cli** interface (by running `au new` command). Please let us know if something was missed.

[aurelia-animator-css.d.ts](https://github.com/aurelia/animator-css/blob/master/dist/aurelia-animator-css.d.ts)
[aurelia-binding.d.ts](https://github.com/aurelia/binding/blob/master/dist/aurelia-binding.d.ts)
[aurelia-bootstrapper.d.ts](https://github.com/aurelia/bootstrapper/blob/master/dist/aurelia-bootstrapper.d.ts)
[aurelia-dependency-injection.d.ts](https://github.com/aurelia/dependency-injection/blob/master/dist/aurelia-dependency-injection.d.ts)
[aurelia-event-aggregator.d.ts](https://github.com/aurelia/event-aggregator/blob/master/dist/aurelia-event-aggregator.d.ts)
[aurelia-framework.d.ts](https://github.com/aurelia/framework/blob/master/dist/aurelia-framework.d.ts)
[aurelia-history.d.ts](https://github.com/aurelia/history/blob/master/dist/aurelia-history.d.ts)
[aurelia-history-browser.d.ts](https://github.com/aurelia/history-browser/blob/master/dist/aurelia-history-browser.d.ts)
[aurelia-loader.d.ts](https://github.com/aurelia/loader/blob/master/dist/aurelia-loader.d.ts)
[aurelia-loader-default.d.ts](https://github.com/aurelia/loader-default/blob/master/dist/aurelia-loader-default.d.ts)
[aurelia-logging.d.ts](https://github.com/aurelia/logging/blob/master/dist/aurelia-logging.d.ts)
[aurelia-logging-console.d.ts](https://github.com/aurelia/logging-console/blob/master/dist/aurelia-logging-console.d.ts)
[aurelia-metadata.d.ts](https://github.com/aurelia/metadata/blob/master/dist/aurelia-metadata.d.ts)
[aurelia-pal.d.ts](https://github.com/aurelia/pal/blob/master/dist/aurelia-pal.d.ts)
[aurelia-pal-browser.d.ts](https://github.com/aurelia/pal-browser/blob/master/dist/aurelia-pal-browser.d.ts)
[aurelia-path.d.ts](https://github.com/aurelia/path/blob/master/dist/aurelia-path.d.ts)
[aurelia-polyfills.d.ts](https://github.com/aurelia/polyfills/blob/master/dist/aurelia-polyfills.d.ts)
[aurelia-route-recognizer.d.ts](https://github.com/aurelia/route-recognizer/blob/master/dist/aurelia-route-recognizer.d.ts)
[aurelia-router.d.ts](https://github.com/aurelia/router/blob/master/dist/aurelia-router.d.ts)
[aurelia-task-queue.d.ts](https://github.com/aurelia/task-queue/blob/master/dist/aurelia-task-queue.d.ts)
[aurelia-templating.d.ts](https://github.com/aurelia/templating/blob/master/dist/aurelia-templating.d.ts)
[aurelia-templating-binding.d.ts](https://github.com/aurelia/templating-binding/blob/master/dist/aurelia-templating-binding.d.ts)
[aurelia-templating-resources.d.ts](https://github.com/aurelia/templating-resources/blob/master/dist/aurelia-templating-resources.d.ts)
[aurelia-templating-router.d.ts](https://github.com/aurelia/templating-router/blob/master/dist/aurelia-templating-router.d.ts)
[aurelia-testing.d.ts](https://github.com/aurelia/testing/blob/master/dist/aurelia-testing.d.ts)

**Comment by AndreyChechel** [comment]
A few packages from the list depend on [lib.es2015.collection.d.ts](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.es2015.collection.d.ts), so implementation could be carrying on once that declaration file is supported. 

## See Also
[#6] Support ECMAScript 2015 declarations is ready.

**Comment by beercsaba** [comment]
Hi Andrey!

Thank You for your effort! 

Your library list is good, but these files may be useful also:
[webpack-plugin](https://github.com/aurelia/webpack-plugin/tree/master/dist/types/*)
[i18n](https://github.com/aurelia/i18n/blob/master/dist/aurelia-i18n.d.ts)
[dialog](https://github.com/aurelia/dialog/tree/master/dist) - with subfolders for different module systems
[fetch-client](https://github.com/aurelia/fetch-client/blob/master/dist/aurelia-fetch-client.d.ts)
[http-client](https://github.com/aurelia/http-client/blob/master/dist/aurelia-http-client.d.ts)
[loader-webpack](https://github.com/aurelia/loader-webpack/tree/master/dist) - with subfolders for different module Systems
[bootstrapper-webpack](https://github.com/aurelia/bootstrapper-webpack/blob/master/dist/aurelia-bootstrapper-webpack.d.ts)
[validate](https://github.com/aurelia/validatejs/blob/master/dist/aurelia-validatejs.d.ts)

and some other d.ts files are [here](https://github.com/aurelia/typings/tree/master/dist)

i hope supporting lib.es2015.collections.d.ts won't be a hard task!

**Comment by beercsaba** [comment]
Hi Andrey!

Thank You for your effort! 

Your library list is good, but these files may be useful also:
[webpack-plugin](https://github.com/aurelia/webpack-plugin/tree/master/dist/types/*)
[i18n](https://github.com/aurelia/i18n/blob/master/dist/aurelia-i18n.d.ts)
[dialog](https://github.com/aurelia/dialog/tree/master/dist) - with subfolders for different module systems
[fetch-client](https://github.com/aurelia/fetch-client/blob/master/dist/aurelia-fetch-client.d.ts)
[http-client](https://github.com/aurelia/http-client/blob/master/dist/aurelia-http-client.d.ts)
[loader-webpack](https://github.com/aurelia/loader-webpack/tree/master/dist) - with subfolders for different module Systems
[bootstrapper-webpack](https://github.com/aurelia/bootstrapper-webpack/blob/master/dist/aurelia-bootstrapper-webpack.d.ts)
[validate](https://github.com/aurelia/validatejs/blob/master/dist/aurelia-validatejs.d.ts)

and some other d.ts files are [here](https://github.com/aurelia/typings/tree/master/dist)

i hope supporting lib.es2015.collections.d.ts won't be a hard task!

**Comment by beercsaba** [comment]
sorry about clcosing and repoening... :S

**Comment by AndreyChechel** [comment]
Hi @beercsaba,

Just want to update you that most of the listed above Aurelia libraries have been supported. Retyped packages for them are already available on NuGet.org:
https://retyped.com/?q=aurelia#search

**Comment by beercsaba** [comment]
Thanks, @AndreyChechel !

i have had no time yet to write thankful  words here...
What  a great work, i can't wait to give it a try!

Should i close this issue now or only after all of the mentioned libraries are supported?

**Comment by AndreyChechel** [comment]
Correct, we'll close the issue once everything mentioned is supported.
